web/src/lib/auth/signinHelper.ts

```diff
@@ -80,30 +81,40 @@ export const onSelfRegister = (
 export const onGuestSignIn = async (
   push: (url: string) => Promise<boolean>,
   pathname: string,
-  dispatch: Dispatch<AuthAction>
+  dispatch: Dispatch<AuthAction>,
+  params?: { encounteredMyNjLinkingError: boolean }
```

Comment:
[sand] Should all of these positional params be included in the new `params` object? It feels weird to me to have competing styles here.

Reply:
I could go either way. The reason for this style is following the "config" pattern that a lot of JS libraries use. Like for example axios, you do 
```
axios.post("/endpoint", data, { configKey: "value" }
```

Where the last argument is an optional object of additional config parameters, but the primary parameters to the function get their top-level importance.